### DISCUSSION TITLE
How to create and use an extension for quarto book

### DISCUSSION THREAD

**Discussion by produnis** [quarto-dev/quarto-cli#10202] [Q&A] [answered]
Hi there, 
I tried to create and embed an extension for a quarto book (pdf) as I did for "simple" pdf documents, but it seems that I have to use other parameters, as all my settings seems to be ignored with quarto book.

I cannot find any Manuals for that...
Thx in advance

**Comment by mcanouil** [comment]
It should work the same way.
Without a concrete example, it's hard to really help.

Did you take a look at the "starter template" from the extension guide? https://quarto.org/docs/extensions/creating.html

**Comment by produnis** [comment]
Hi there, I uploaded a minimal example to this specific branch:
https://github.com/produnis/minimal_examples/tree/quarto-book-extension

This version renders with my extension "`mybook`". As you can see, all the settings in my extensions yaml are ignored (german language, title page, etc)

In `_quarto2.yaml`, I added all settings from the extension directly into the `pdf:` specs. If you use this for rendering, you can see how all settings are executed as expected (title page, german listing "Inhaltsverzeichnis", etc).

Btw: are you at useR2024 in Salzburg ?

**Comment by produnis** [comment, marked as answer]
lol, I had the wrong title in my extension yaml...
never mind and sorry,
working now

but worth noting:
if you give a wrong folder name in `format:`, quarto will return an error. If you give  the correct folder name, like `format: mybook-pdf`, it will return no error, even if the name in `_extension.yaml`'s `title` field does not match the extension's folder name.   
In my case, it ignored all settings, as I had the wrong title in my `extension.yaml`.
